gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -504,6 +504,155 @@ def _get_variant_calls(self, record, formats):
+
+    p_read, c_write = os.pipe()
+    c_read, p_write = os.pipe()
+
+    processid = os.fork()
+    if processid:
+      os.close(c_read)
+      os.close(c_write)
+      into_pysam = os.fdopen(p_read)
+      self._to_child = os.fdopen(p_write, 'w')
+      self._vcf_reader = libcbcf.VariantFile(into_pysam, 'r')
+    else:
+      os.close(p_read)
+      os.close(p_write)
+      from_parent = os.fdopen(c_read)
+      to_pysam = os.fdopen(c_write, 'w')
+      # Feed the data with a break line, since we read data line by line - if
+      # we would use the conventional ``read()`` method, both pipes would lock,
+      # while waiting for an EOF signal.
+      for header in header_lines:
+        to_pysam.write(header.strip() + '\n')
+      to_pysam.flush()
+
+      while 1:
+        text_line = from_parent.readline()
+        if not text_line:
+          break
+        to_pysam.write(text_line)
+        to_pysam.flush()
+
+      from_parent.close()
+      to_pysam.close()
+      sys.exit(0)
+
+  def _get_variant(self, data_line):
+    try:
+      self._to_child.write(data_line + '\n')
+      self._to_child.flush()
+      return self._convert_to_variant(next(self._vcf_reader))
+    except (MemoryError, IOError) as e:
+      logging.warning('VCF record read failed in %s for line %s: %s',
+                      self._file_name, data_line, str(e))
+      return MalformedVcfRecord(self._file_name, data_line, str(e))
+
+  def _convert_to_variant(self, record):
+    # type: (libcbcf.VariantRecord) -> Variant
+    """Converts the PySAM record to a :class:`Variant` object.
+
+    Args:
+      record: An object containing info about a variant.
+    Returns:
+      A :class:`Variant` object from the given record.
+    Raises:
+      ValueError: if ``record`` is semantically invalid.
+    """
+
+    return Variant(
+        reference_name=record.chrom,
+        start=record.start,
+        end=record.stop,
+        reference_bases=record.ref,
+        alternate_bases=list(record.alts) if record.alts else [],
+        names=record.id.split(';') if record.id else [],
+        quality=float(record.qual) if record.qual else None,
+        filters=['PASS'] if not record.filter.keys() else record.filter.keys(),
+        info=self._get_variant_info(record),
+        calls=self._get_variant_calls(record.samples))
+
+  def _get_variant_info(self,
+                        record  # type: libcbcf.VariantRecord
+                       ):
+    info = {}
+    for k, v in record.info.iteritems():
+      if k != END_INFO_KEY:
+        if isinstance(v, tuple):
+          info[k] = list(v)
+        else:
+          info[k] = v
+
+    return info
+
+  def _get_variant_calls(self, samples):
+    # type: (libcvcf.VariantRecordSamples) -> List[VariantCall]
+    calls = []
+
+    for (name, sample) in samples.iteritems():
+      if sample.phased:
+        phaseset = DEFAULT_PHASESET_VALUE
+      else:
+        phaseset = None
```

Comment:
simplify to `phaseset = DEFAULT_PHASESET_VALUE if sample.phased else None` if it can fit into one line